Dockerfile

```diff
@@ -1,74 +1,142 @@
-FROM python:3.8
+FROM golang:1.13 as go-builder
+ARG IMPOSM_REPO="https://github.com/omniscale/imposm3.git"
+ARG IMPOSM_VERSION="v0.10.0"
 
-WORKDIR /usr/src/app
+# Build imposm
+RUN set -eux ;\
+    go version ;\
+    go get github.com/tools/godep ;\
+    mkdir /build-bin ;\
+    \
+    DEBIAN_FRONTEND=noninteractive apt-get update ;\
+    DEBIAN_FRONTEND=noninteractive apt-get install  -y --no-install-recommends \
+        `# installing imposm dependencies` \
+        libgeos-dev \
+        libleveldb-dev \
+        ;\
+    /bin/bash -c 'echo ""; echo ""; echo "##### Build imposm3 -- https://github.com/osm2vectortiles/imposm3"' >&2 ;\
+    #
+    # get and build specific version of imposm
+    git clone --quiet --depth 1 $IMPOSM_REPO -b $IMPOSM_VERSION \
+        $GOPATH/src/github.com/omniscale/imposm3 ;\
+    cd $GOPATH/src/github.com/omniscale/imposm3 ;\
+    make build ;\
+    # Older imposm executable was called imposm3 - rename it to the common name "imposm"
+    ( [ -f imposm ] && mv imposm /build-bin/imposm || mv imposm3 /build-bin/imposm )
+
+
+# Build osmborder
+FROM python:3.8 as c-builder
+ARG OSMBORDER_REV=e3ae8f7a2dcdcd6dc80abab4679cb5edb7dc6fa5
+
+RUN set -eux ;\
+    mkdir /build-bin ;\
+    DEBIAN_FRONTEND=noninteractive apt-get update ;\
+    DEBIAN_FRONTEND=noninteractive apt-get install  -y --no-install-recommends \
+        `# installing osmborder dependencies` \
+        build-essential \
+        ca-certificates \
+        cmake \
+        git \
+        libosmium2-dev \
+        zlib1g-dev \
+        ;\
+    /bin/bash -c 'echo ""; echo ""; echo "##### Building osmborder -- https://github.com/pnorman/osmborder"' >&2 ;\
+    git clone https://github.com/pnorman/osmborder.git /usr/src/osmborder ;\
+    cd /usr/src/osmborder ;\
+    git checkout ${OSMBORDER_REV?} ;\
```

Comment:
@nyurik can you explain, why there is sometimes construction like `${VAR?}` and sometimes like `${VAR:?}`?